packages/dashboard/src/components/Widget/Widget.tsx

```diff
@@ -71,7 +72,24 @@ const Widget = ({
 
   const deleteWidget = () => {
     if (!widgetConfig) return
+    // if last widget left in row remove whole row on delete
+    if (widgetConfig.colIdx === 0) {
+      let newVisualizations = { ...config.visualizations }
+      const rows = config.rows
+      const rowIdx = widgetConfig.rowIdx
+
+      //delete the instantiated widgets
+      if (rows[rowIdx] && rows[rowIdx].columns && rows[rowIdx].columns.length && config.visualizations) {
+        rows[rowIdx].columns.forEach(column => {
+          if (column.widget) {
+            delete newVisualizations[column.widget]
+          }
+        })
+      }
 
+      config.rows.splice(rowIdx, 1) // delete the row
```

Comment:
I think the functionality should be that you delete the column but leave a placeholder there, a single column so that you can still add a widget there if you want. 

say for example you put a chart widget when you meant to put a table widget, it would be annoying if the row disappeared when you deleted the mistaken chart widget. you'd have to add a new row and if you have a lot of rows, move it all the way to the top.